pkg/kvstore/etcd.go

```diff
@@ -852,15 +852,16 @@ reList:
 			if traceEnabled {
 				scopedLog.Debugf("Emitting EventTypeDelete event for %s", k)
 			}
-
-			queueStart := spanstat.Start()
-			w.Events <- event
-			trackEventQueued(scope, EventTypeDelete, queueStart.End(true).Total())
+			if ok := w.emit(ctx, scope, event); !ok {
+				return
+			}
```

Comment:
This return statement does not have any effect, as it only returns from the internal function. One possibility could be to propagate the return value to `localCache.RemoveDeleted`, or alternatively add a subsequent select statement and exit if any of the context/stopChan conditions is met before possibly emitting the `listDone` event.

Reply:
I chose to propagate the return value